src/file_io.jl

```diff
@@ -74,7 +74,9 @@ Creates a JLSOFile with the specified data and kwargs and writes it back to the
 """
 save(io::IO, data; kwargs...) = write(io, JLSOFile(data; kwargs...))
 save(io::IO, data::Pair...; kwargs...) = save(io, Dict(data...); kwargs...)
-save(path::String, args...; kwargs...) = open(io -> save(io, args...; kwargs...), path, "w")
+function save(path::Union{AbstractPath, AbstractString}, args...; kwargs...)
```

Comment:
FWIW, there's an `@compat` macro in FilePaths.jl that will create a wrapper string version for you.

https://github.com/rofinn/FilePaths.jl/blob/master/src/compat.jl

Downside is that it adds a couple extra dependencies.

Reply:
yah, the extra dependencies is why i'm not using. If we needed many more changes like this and/or had more complex changes, like String/Path in many positions, then it _may_ be worth it. But i don't think it is worth it for just these two cases